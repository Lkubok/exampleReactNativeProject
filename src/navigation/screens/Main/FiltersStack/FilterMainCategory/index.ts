import Screen from './FilterMainCategory';

export default Screen;
