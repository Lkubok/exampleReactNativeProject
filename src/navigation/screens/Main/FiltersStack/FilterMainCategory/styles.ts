import { StyleSheet } from 'react-native';

import { variables } from 'res';

export default StyleSheet.create({
  container: {
    backgroundColor: variables.colors.white,
    paddingHorizontal: variables.spacing.base,
    paddingTop: variables.spacing.base,
  },
});
