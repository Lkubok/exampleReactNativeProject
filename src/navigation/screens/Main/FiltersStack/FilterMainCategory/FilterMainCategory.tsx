import React, { FC, useMemo, useCallback } from 'react';
import { ScrollView } from 'react-native';
import { useSelector, useDispatch } from 'react-redux';
import { map } from 'lodash';
import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';

import { CategoriesPicker, Navbar } from 'components';
import { useTranslations } from 'hooks';
import { selectors as filterSelectors, actions as filterActions } from 'store/filters';
import { actions as newOfferActions, selectors as newOfferSelector } from 'store/newOffer';
import { selectors as basicDataSelectors } from 'store/basicData';
import { CategoryItemType } from 'models/types';
import { FilterStackParamList, screenNames } from 'navigation';

import styles from './styles';

const FilterMainCategory: FC = () => {
  const { i18n } = useTranslations();
  const mainSelectedCategoryId = useSelector(filterSelectors.getMainCategorySelectedFilterId);
  const newOfferMainSelectedCategoryId = useSelector(newOfferSelector.getNewOfferCategoryId);
  const availableMainCategories = useSelector(basicDataSelectors.getCategoriesList);
  const dispatch = useDispatch();
  const { goBack } = useNavigation();
  const { params } = useRoute<
    RouteProp<FilterStackParamList, typeof screenNames.MainFlow.ModalFilterStack.FiltersMainCategory>
  >();

  const mainAvailableFilters: CategoryItemType[] = useMemo(
    () =>
      map(availableMainCategories, (category) => {
        const selectedCategoryId =
          params?.mode === 'newOffer' ? newOfferMainSelectedCategoryId : mainSelectedCategoryId;
        return {
          id: category.id,
          name: category.name,
          checked: selectedCategoryId === category.id,
        };
      }),
    [availableMainCategories, mainSelectedCategoryId, newOfferMainSelectedCategoryId, params?.mode],
  );

  const handleMainCategorySelect = useCallback(
    (id: number | string) => {
      if (typeof id === 'number') {
        if (params?.mode === 'marketDashboard') {
          dispatch(filterActions.setMainCategoriesFilterId(id));
          dispatch(filterActions.clearSubcategoriesFilters());
        }
        if (params?.mode === 'newOffer') {
          dispatch(newOfferActions.setMainCategoryID(id));
          dispatch(newOfferActions.clearSubcategory());
        }
        goBack();
      }
    },
    [dispatch, goBack, params.mode],
  );

  return (
    <>
      <Navbar withLinkBack center={i18n.t('navbar_main_category')} inModal />
      <ScrollView style={styles.container}>
        <CategoriesPicker onPress={handleMainCategorySelect} items={mainAvailableFilters} singleSelect />
      </ScrollView>
    </>
  );
};

export default FilterMainCategory;
