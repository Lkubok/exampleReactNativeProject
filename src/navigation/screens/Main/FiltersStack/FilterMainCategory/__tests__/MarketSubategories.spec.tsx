import React from 'react';
import { act, MockedNavigator, originRender } from 'utils/testUtils';

import FilterMainCategory from '../FilterMainCategory';

jest.useFakeTimers();

describe('FilterMainCategory screen', () => {
  beforeEach(() => {});

  it('should match snapshot', () => {
    const snapshot = originRender(<MockedNavigator component={FilterMainCategory} />).toJSON();
    // eslint-disable-next-line no-void
    void act(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
