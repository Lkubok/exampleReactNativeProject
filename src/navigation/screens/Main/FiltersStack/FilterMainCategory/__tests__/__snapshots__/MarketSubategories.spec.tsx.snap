// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMainCategory screen should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 44,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 0,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "overflow": undefined,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "[7m[38;2;0;0;0m#000[39m[27m",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "flex": 1,
                    "overflow": "hidden",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "[7m[38;2;248;248;248m#f8f8f8[39m[27m",
                          "borderBottomColor": "[7m[38;2;196;196;198m#c4c4c6[39m[27m",
                          "borderBottomWidth": 0.5,
                          "borderTopEndRadius": 12,
                          "borderTopStartRadius": 12,
                          "flexDirection": "row",
                          "minHeight": 54,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "flex-start",
                            "flexGrow": 1,
                            "justifyContent": "center",
                            "paddingLeft": 16,
                            "zIndex": 1,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          hitSlop={
                            Object {
                              "bottom": 24,
                              "left": 24,
                              "right": 24,
                              "top": 24,
                            }
                          }
                          onClick={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <SvgMock
                            fill="#ea3336"
                            height={24}
                            width={24}
                          />
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                            "left": 0,
                            "minHeight": 54,
                            "position": "absolute",
                            "right": 0,
                            "width": "100%",
                            "zIndex": 0,
                          }
                        }
                      >
                        <View>
                          <Text
                            style={
                              Object {
                                "color": "[7m[38;2;0;0;0m#000000[39m[27m",
                                "fontFamily": "SFCompactText-Semibold",
                                "fontSize": 17,
                                "letterSpacing": -0.41,
                              }
                            }
                          >
                            Main category
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "flexGrow": 1,
                            "justifyContent": "flex-end",
                            "paddingRight": 16,
                            "zIndex": 1,
                          }
                        }
                      />
                    </View>
                    <RCTScrollView
                      style={
                        Object {
                          "backgroundColor": "[7m[38;2;255;255;255m#ffffff[39m[27m",
                          "paddingHorizontal": 12,
                          "paddingTop": 12,
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "marginBottom": 32,
                              "marginTop": 4,
                            }
                          }
                        />
                      </View>
                    </RCTScrollView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </View>
</RNCSafeAreaProvider>
`;
